### PR TITLE
feat(perf-issues): initialize giant http payload detector

### PR DESCRIPTION
**Pull request by DominikB2014**
Initializes the giant http payload detector. This is a simple detector that just compares the payload size to a threshold (10mb).
We won't be able to detect issues from this as the SDK changes to get the payload size for fetch and xhr is not in place, but putting this in place to have it ready.

### CODE REVIEW

File: tests/sentry/utils/performance_issues/test_large_http_payload_detector.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    "parent_span_ids": [],
+                    "cause_span_ids": [],
+                    "offender_span_ids": "bbbbbbbbbbbbbbbb",
+                    "op": "http",
+                },
+                evidence_display=[],
+            )
+        ]
+
+    def test_does_not_detect_large_asset(self):
```
Comment (gggritso) [inline tests/sentry/utils/performance_issues/test_large_http_payload_detector.py:66]: I think this test needs a better name, I can't quite figure out what exactly it's testing by looking at it

File: src/sentry/utils/performance_issues/detectors/large_payload_detector.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            evidence_display=[],
+            evidence_data={
+                "parent_span_ids": [],
+                "cause_span_ids": [],
+                "offender_span_ids": offender_span_ids,
+                "op": "http",
+            },
+        )
+
+    def _is_eligible_http_span(self, span: Span) -> bool:
```
Comment (gggritso) [inline src/sentry/utils/performance_issues/detectors/large_payload_detector.py:59]: Nit, but in other detectors this is a class method called `is_span_eligible`, would be good to keep it consistent!

**Review by gggritso — APPROVED** [review]
LGTM overall, just a few nits!

File: tests/sentry/utils/performance_issues/test_large_http_payload_detector.py (hunk 1 above)
Comment (DominikB2014) [inline tests/sentry/utils/performance_issues/test_large_http_payload_detector.py:66]:
```suggestion
    def test_does_not_issue_if_url_is_an_asset(self):
```

File: src/sentry/utils/performance_issues/detectors/large_payload_detector.py (hunk 2 above)
Comment (DominikB2014) [reply src/sentry/utils/performance_issues/detectors/large_payload_detector.py]: done!

**Pull request merged**